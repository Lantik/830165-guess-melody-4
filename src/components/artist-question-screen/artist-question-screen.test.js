import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen.jsx';

const mock = {
  question: {
    type: `artist`,
    song: {
      src: ``,
      artist: `one`
    },
    answers: [{
      picture: `pic-one`,
      artist: `one`,
    }, {
      picture: `pic-two`,
      artist: `two`,
    }, {
      picture: `pic-three`,
      artist: `three`,
    }
    ]
  }
};

it(`<ArtistQuestionScreen/> rendered correctly`, () => {
  const {question} = mock;
  const tree = renderer
    .create(<ArtistQuestionScreen
      question={question}
      onAnswer={()=>{}}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
