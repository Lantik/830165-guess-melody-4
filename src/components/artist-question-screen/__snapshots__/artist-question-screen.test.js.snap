// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistQuestionScreen/> rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <div
        className="player"
      >
        src: 
        https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg
        ; id: 
        0
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        checked={false}
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="one"
          className="artist__picture"
          src="pic-one"
        />
        one
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        checked={false}
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="two"
          className="artist__picture"
          src="pic-two"
        />
        two
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        checked={false}
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="three"
          className="artist__picture"
          src="pic-three"
        />
        three
      </label>
    </div>
  </form>
</section>
`;
